WordPress/src/main/java/org/wordpress/android/ui/mediapicker/loader/GifMediaDataSource.kt

```diff
@@ -0,0 +1,131 @@
+package org.wordpress.android.ui.mediapicker.loader
+
+import android.content.Context
+import android.net.Uri
+import com.tenor.android.core.constant.AspectRatioRange
+import com.tenor.android.core.constant.MediaCollectionFormat
+import com.tenor.android.core.constant.MediaFilter
+import com.tenor.android.core.model.impl.Result
+import com.tenor.android.core.network.ApiClient
+import com.tenor.android.core.network.IApiClient
+import com.tenor.android.core.response.impl.GifsResponse
+import org.wordpress.android.R
+
+import org.wordpress.android.ui.mediapicker.MediaItem
+import org.wordpress.android.ui.mediapicker.MediaItem.Identifier.GifMediaIdentifier
+import org.wordpress.android.ui.mediapicker.MediaType.IMAGE
+import org.wordpress.android.ui.mediapicker.loader.MediaSource.MediaLoadingResult
+import org.wordpress.android.ui.mediapicker.loader.MediaSource.MediaLoadingResult.Failure
+import org.wordpress.android.ui.mediapicker.loader.MediaSource.MediaLoadingResult.Success
+import org.wordpress.android.viewmodel.gif.provider.GifProvider.GifRequestFailedException
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class GifMediaDataSource
+@Inject constructor(
+    private val context: Context,
+    private val tenorClient: IApiClient
+) : MediaSource {
+    private var nextPosition: Int = 0
+    private var items = mutableListOf<MediaItem>()
```

Comment:
It can either be `var` or mutable list, you don't need both. Either rewrite the variable or use `clear` and `addAll` on a mutable list.

Reply:
ahah, totally right 😄 ! Changed [here](7a5d7f94cc965e6135ae82d67391bf8d96a46c1b) 👍